### ISSUE TITLE
NotEqual works incorrectly

### ISSUE THREAD

**Issue by tesla-cat** [weaviate/weaviate#3743]
## Equal

```py
.with_where(
            {
                "path": ["uid"],
                "operator": "Equal",
                "valueText": idea["uid"],
            }
        )
```

<img width="335" alt="image" src="https://github.com/weaviate/weaviate/assets/28215638/901d1383-76f7-41fc-b7a0-c49fb629ccb3">

## NotEqual

**the circled item should not appear**

```py
.with_where(
            {
                "path": ["uid"],
                "operator": "NotEqual",
                "valueText": idea["uid"],
            }
        )
```

<img width="363" alt="image" src="https://github.com/weaviate/weaviate/assets/28215638/2d39fcbd-c37e-4770-bf41-4c777f9d137d">

**Comment by LorenzBuehmann** [comment]
isn't that the same issue as here?
https://github.com/weaviate/weaviate/issues/3599

**Comment by parkerduckworth** [comment]
@LorenzBuehmann yes, I believe this is a related issue.  #3319 has been implemented with #4122, and will be included with the upcoming 1.24 release.

Once it's out, I recommend upgrading and retrying your use case @tesla-cat
